tool/src/java/org/sakaiproject/gradebookng/tool/panels/AddGradeItemPanel.java

```diff
@@ -14,14 +27,82 @@
 
 	private static final long serialVersionUID = 1L;
 
-	public AddGradeItemPanel(String id) {
+	public AddGradeItemPanel(String id, List<CategoryDefinition> categories, final ModalWindow window) {
```

Comment:
Have a look at the other panels. If you inject the business service in you can do this logic inside there. then have a getter in the gradebook page for the modal window.  Passing stuff into the constructor isnt safe for this.